Use 'six' module to make the software Python3-compatible. Automatical…

PaxG opened the pull request:
Use 'six' module to make the software Python3-compatible. 
Automatically creates slow mountpoint if directory does not exist yet. 
Added -d shortcut for --debug. 
Code to be tested under Python2

Commit:
Use 'six' module to make the software Python3-compatible. Automatically creates slow mountpoint if directory does not exist yet. Added -d shortcut for --debug. Code to be tested under Python2

nirs commented on slowfs.py:
```diff
@@ -15,14 +15,19 @@
-import fuse
-
+import fuse      # Filesystem in UserSpacE
+import six       # Python 2 and 3 Compatibility Library (2*3=6)
```
I don't think these comments are needed, and they are not relevant for supporting python 3.

Also aligning the comments is not compatible with pep8. While we don't enforce pep8 yet, I don't like to introduce incompatible code.

Please check your changes with flake8  to make sure new code conforms to pep8.

nirs commented on slowfs.py:
```diff
@@ -15,14 +15,19 @@
+            os.mkdir(args.mountpoint)
+        except Exception as e:
+            print("Error creating mountpoint {}: {}".format(args.mountpoint, e))
+            sys.exit(1)
```
This change is not related to supporting python 3, and I don't think it is needed. This may create the mountpoint with incorrect ownership and permissions.

Please send this change in another PR if you like.

nirs commented on slowfs.py:
```diff
@@ -33,7 +38,7 @@ def parse_args(args):
     parser.add_argument('-c', '--config',
                         help='path to configuration file')
-    parser.add_argument('--debug', action='store_true',
+    parser.add_argument('-d', '--debug', action='store_true',
```
Good change, but also not related to python 3 - please send another PR for rthis.

nirs commented on slowfs.py:
```diff
@@ -64,7 +69,9 @@ def set(self, name, value):
-            exec f in d, d
+            # exec f in d, d   # Not Python3 compatible syntax
+            # f is now a io.TextIOWrapper object, whereas "exec" needs a string, bytes or code object => add .read() method
+            six.exec_(f.read(),d)   # Execute code in the scope of globals and locals. code can be a string or a code object. 
```
Using six.exec_ is fine and should work on both python 2 and 3.

However using open() is not correct, we should use:

    io.open(self._path, "rb")

Which behave exactly the same in python 2 and 3, opening the file in binary mode.

Again the commented code should be removed, and the comment is not needed and should not be aligned in this way. If you think this needs some documentation, please add a docstring to the function.

nirs commented on slowfs.py:
```diff
@@ -98,7 +105,8 @@ def _run(self):
             msg, sender = self.sock.recvfrom(1024)
-        except socket.error, e:
+        # except socket.error, e:    Not Python3 compatible syntax
+        except socket.error as e:
```
Looks good, please remove the commented code.

nirs reviewed (changes requested):
Thanks Pascal!

I hope you can remove the unrelated changes, and send separate PRs for them.

If not, I can take the python 3 related changes from your PR.

Commit:
Follow good advices from Nir: removed changes not related to Python3 compatibility, used flake8 to chech pep8 compatibility, removed useless comments, replaced 'open' by io.open(..., rb) for Python 2/3 compatibility

PaxG commented:
Thanks for your review! I removed changes not related to Python 2/3 compatibility, removed useless comments and used flake8 to check code style. New changes are pushed on the python3 branch.

nirs reviewed (approved):
The patch looks fine now, thanks!

nirs commented:
Thanks Pascal, I merge your PR with minor changes in 9e835eacc0e8596d725f9c1ac55072ae7f24b58e.

- Separate standard library modules (e.g. io) from 3rd party modules (e.g. six)
- Rewrite the commit message
